emrun.py

```diff
@@ -400,6 +420,35 @@ def kill_browser_process():
     # Clear the process name to represent that the browser is now dead.
     processname_killed_atexit = ''
 
+  delete_emrun_safe_firefox_profile()
+
+
+# Heuristic that attempts to search for the browser process IDs that emrun spawned.
+# This depends on the assumption that no other browser process IDs have been spawned
+# during the short time perioed between the time that emrun started, and the browser
+# process navigated to the page.
+# This heuristic is needed because all modern browsers are multiprocess systems -
+# starting a browser process from command line generally launches just a "stub" spawner
+# process that immediately exits.
+def detect_browser_processes():
+  global previous_browser_processes, current_browser_processes, navigation_has_occurred
+  logv('First navigation occurred. Identifying currently running browser processes')
+  running_browser_processes = list_processes_by_name(browser_exe)
+
+  def pid_existed(pid):
+    for proc in previous_browser_processes:
+      if proc['pid'] == pid:
+        return True
+    return False
+
+  for p in running_browser_processes:
+    logv('Detected running browser process id: ' + str(p['pid']) + ', existed already at emrun startup? ' + str(pid_existed(p['pid'])))
+
+  current_browser_processes = list(filter(lambda x: not pid_existed(x['pid']), running_browser_processes))
```

Comment:
I guess this is only moved code but could be clearer as:

`current_browser_processes = [p for p in running_browser_processes if not pid_existed(p['pid'])]`